src/main/java/duke/storage/TaskListEncoder.java

```diff
@@ -0,0 +1,21 @@
+package duke.storage;
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.task.TaskList;
+/**
+ * Encodes a list of task and saves it in a text file.
+ */
+
+public class TaskListEncoder {
+    /**
+     * Encodes the list of tasks to be saved into a text file.
+     * @param toSave the list of tasks to be saved
+     * @return a list of strings from the original list of tasks
+     */
+    public static List<String> encodeTaskList(TaskList toSave) {
+        final List<String> encodedTasks = new ArrayList<String>();
+        toSave.getAllTasks().forEach(task -> encodedTasks.add(task.toString()));
```

Comment:
The use of the forEach() method to adbstract the code is great; I really like hope succint your code is!